contracts/pcv/PCVGuardian.sol

```diff
@@ -0,0 +1,181 @@
+        );
+        _;
+    }
+
+    // ---------- Read-Only API ----------
+
+    /// @notice returns true if the the provided address is a valid destination to withdraw funds from
+    /// @param pcvDeposit the pcvDeposit address to check if whitelisted
+    function isWhitelistAddress(address pcvDeposit)
+        public
+        view
+        override
+        returns (bool)
+    {
+        return whitelistAddresses.contains(pcvDeposit);
+    }
+
+    /// @notice returns all whitelisted pcvDeposit addresses
+    function getWhitelistAddresses()
+        public
+        view
+        override
+        returns (address[] memory)
+    {
+        return whitelistAddresses.values();
+    }
+
+    // ---------- Governor-Only State-Changing API ----------
+
+    /// @notice governor-only method to whitelist a pcvDesposit address to withdraw funds from
+    /// @param pcvDeposit the address to set as safe
+    function addWhitelistAddress(address pcvDeposit)
+        external
+        override
+        onlyGovernor
+    {
+        _addWhitelistAddress(pcvDeposit);
+    }
+
+    /// @notice batch version of setWhiteListaddress
+    /// @param _whitelistAddresses the addresses to whitelist, as calldata
+    function addWhitelistAddresses(address[] calldata _whitelistAddresses)
+        external
+        override
+        onlyGovernor
+    {
+        // improbable to ever overflow
+        unchecked {
+            require(_whitelistAddresses.length != 0, "empty");
+            for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+                _addWhitelistAddress(_whitelistAddresses[i]);
+            }
+        }
+    }
+
+    // ---------- Governor-Or-Guardian-Only State-Changing API ----------
+
+    /// @notice governor-or-guardian-only method to un-whitelist a pcvDesposit address to withdraw funds from
+    /// @param pcvDeposit the address to un-set as safe
+    function removeWhitelistAddress(address pcvDeposit)
+        external
+        override
+        onlyGuardianOrGovernor
+    {
+        _removeWhitelistAddress(pcvDeposit);
+    }
+
+    /// @notice batch version of unsetWhitelistAddress
+    /// @param _whitelistAddresses the addresses to un-whitelist
+    function removeWhitelistAddresses(address[] calldata _whitelistAddresses)
+        external
+        override
+        onlyGuardianOrGovernor
+    {
+        // improbable to ever overflow
+        unchecked {
+            require(_whitelistAddresses.length != 0, "empty");
+            for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+                _removeWhitelistAddress(_whitelistAddresses[i]);
+            }
+        }
+    }
+
+    /// @notice governor-or-guardian-or-pcv-guard method to withdraw funds from a pcv deposit, by calling the withdraw() method on it
+    /// @param pcvDeposit the address of the pcv deposit contract
+    /// @param amount the amount to withdraw
+    function withdrawToSafeAddress(address pcvDeposit, uint256 amount)
```

Comment:
Currently you will only be able to withdraw funds from the PCVDeposit denominated in the underlying.

Do you want this contract to also be able to withdraw other ERC20 tokens that the PCVDeposit may have earnt - yield/reward tokens? Would need another method like withdrawERC20ToSafeAddress which takes a token argument